azkaban-common/src/main/java/azkaban/scheduler/ScheduleManager.java

```diff
@@ -238,4 +246,30 @@ public void loadTriggerFromProps(final Props props) throws ScheduleManagerExcept
   public String getTriggerSource() {
     return SIMPLE_TIME_TRIGGER;
   }
+
+  /**
+   * fetch from TriggerManager to see if the desired trigger is being updated or not.
+   * if updated, update schedule metadata;
+   * if not being updated, return the original schedule.
+   *
+   * @param s, original schedule
+   * @return latest schedule metadata
+   * */
+  private Schedule updateSingleSchedule(@NotNull Schedule s) throws ScheduleManagerException {
+    s.lock();
+    try {
+      Optional<Schedule> updatedSchedule = loader.loadUpdateSchedule(s);
+      if (updatedSchedule.isPresent()) {
+        if (s.getStatus().equals(TriggerStatus.EXPIRED.toString())) {
+          onScheduleExpire(s);
+        } else {
+          internalSchedule(s);
+        }
+        return updatedSchedule.get();
+      }
+      return s;
+    } finally {
+      s.unlock();
```

Comment:
Not super familiar with Java `finally`, I think move `return s;` to the end of method is more easy to read

Reply:
If anything happens before "returnn s", for example an exception happens. this method might exit but this lock would never release. So in java, it's recommended to use "finally" keywords to release lock.

Reply:
in current setting, if no exception, it will unlock and then return? or it will return but also unlock?

Reply:
yes, using "finally" means after the "try" it will whatever unblock. If it successfully executes return it will return.